node/core/pvf/src/worker_common.rs

```diff
@@ -0,0 +1,280 @@
+			.args(extra_args)
+			.arg(socket_path.as_ref().as_os_str())
+			.stdout(async_process::Stdio::piped())
+			.kill_on_drop(true)
+			.spawn()?;
+
+		let stdout = child
+			.stdout
+			.take()
+			.expect("the process spawned with piped stdout should have the stdout handle");
+
+		Ok(WorkerHandle {
+			child,
+			stdout,
+			// We don't expect the bytes to be ever read. But in case we do, we should not use a buffer
+			// of a small size, because otherwise if the child process does return any data we will end up
+			// issuing a syscall for each byte. We also prefer not to do allocate that on the stack, since
+			// each poll the buffer will be allocated and initialized (and that's due poll_read takes &mut [u8]
+			// and there are no guarantees that a `poll_read` won't ever read from there even though that's
+			// unlikely).
+			//
+			// OTOH, we also don't want to be super smart here and we could just afford to allocate a buffer
+			// for that here.
+			drop_box: vec![0; 8192].into_boxed_slice(),
+		})
+	}
+
+	/// Returns the process id of this worker.
+	pub fn id(&self) -> u32 {
+		self.child.id()
+	}
+}
+
+impl futures::Future for WorkerHandle {
+	type Output = ();
+
+	fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+		let me = self.project();
+		match futures::ready!(AsyncRead::poll_read(me.stdout, cx, &mut *me.drop_box)) {
+			Ok(0) => {
+				// 0 means EOF means the child was terminated. Resolve.
+				Poll::Ready(())
+			}
+			Ok(_bytes_read) => {
+				// weird, we've read something. Pretend that never happened and reschedule ourselves.
+				cx.waker().wake_by_ref();
+				Poll::Pending
+			}
+			Err(_) => {
+				// The implementation is guaranteed to not to return WouldBlock and Interrupted. This
+				// leaves us with a legit errors which we suppose were due to termination.
+				Poll::Ready(())
+			}
+		}
+	}
+}
+
+impl fmt::Debug for WorkerHandle {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		write!(f, "WorkerHandle(pid={})", self.id())
+	}
+}
+
+/// Convert the given path into a byte buffer.
+pub fn path_to_bytes(path: &Path) -> &[u8] {
+	// Ideally, we take the OsStr of the path, send that and reconstruct this on the other side.
+	// However, libstd doesn't provide us with such an option. There are crates out there that
+	// allow for extraction of a path, but TBH it doesn't seem to be a real issue.
+	//
+	// However, should be there reports we can incorporate such a crate here.
+	path.to_str().expect("non-UTF-8 path").as_bytes()
+}
+
+/// Interprets the given bytes as a path. Returns `None` if the given bytes do not constitute a
+/// a proper utf-8 string.
+pub fn bytes_to_path(bytes: &[u8]) -> Option<PathBuf> {
+	let str_buf = std::str::from_utf8(bytes).ok()?;
+
+	match PathBuf::from_str(&str_buf) {
+		Ok(path) => Some(path),
+		Err(never) => {
+			// std::convert::Infallible doesn't allow for irrefutable patterns for some reason.
+			match never {}
+		}
+	}
```

Comment:
```suggestion
	PathBuf::from_str(&str_buf).ok()
```

?

Reply:
Yeah I wanted to emphasize that parsing `PathBuf` doesn't fail and introduced this poor man's `Result::into_ok` (which is still nightly only). 

I don't mind changing that, but I would prefer `expect` here rather than `ok` tho

Reply:
Why not just use `PathBuf::from`?

Reply:
ah damn, of course!